app/javascript/src/components/Dashboard/Notes/Pane/Form.jsx

```diff
@@ -3,24 +3,30 @@ import React from "react";
 import { Formik, Form as FormikForm } from "formik";
 import { Button, Pane, Toastr } from "neetoui";
 import { Input, Textarea, Select } from "neetoui/formik";
+import * as R from "ramda";
 import { useTranslation } from "react-i18next";
 
-import notesApi from "apis/notes";
-
 import {
   NOTES_FORM_VALIDATION_SCHEMA,
   CONTACT_SELECT_OPTIONS,
   TAG_SELECT_OPTIONS,
 } from "../constants";
 
-const Form = ({ onClose, refetch, note, isEdit, createNewNote }) => {
+const Form = ({ onClose, note, isEdit, createNewNote, updateEditNote }) => {
   const { t } = useTranslation();
 
   const handleSubmit = async values => {
     try {
       if (isEdit) {
-        await notesApi.update(note.id, values);
-        refetch();
+        updateEditNote(prevNotes => {
+          const filteredNotes = R.reject(
+            prevNote => prevNote.id === note.id,
+            prevNotes
+          );
+
+          return [...filteredNotes, { ...values, id: note.id }];
+        });
```

Comment:
You could compose the whole thing with Ramda itself using merge{Left/Rirhgt} and reject. Since they're curried you could directly pass the functions itself.

Trying to say that if you are using Ramda then go all in. Here I think a simple builtin `filter` function would do the job. Ramda seems very forced.

Reply:
I'll replace it with the filter function for the time being since I'm not 100% clear on Ramda yet.